### PR TITLE
[4.0] Finder pagination

### PR DESCRIPTION
**Pull request by sakiss**
Pull Request for Issue # .

### Summary of Changes
Currently the item's limit for the Finder's result is being read from the Joomla global config.
This allows to set a pagination limit for the Finder's results, in the respective menu item.

### Testing Instructions
Apply the patch, go to the menu item for the finder's results > Advanced >  Select number of items per page
There set the number of items you want.

### Actual result BEFORE applying this Pull Request
Getting the items per page from the Joomla global configuration.

### Expected result AFTER applying this Pull Request
Getting the items per page from the finder's menu item's configuration.

### Documentation Changes Required
DNK

### CODE REVIEW

File: components/com_finder/tmpl/search/default.xml [hunk 1]
```diff
@@ -150,6 +150,22 @@
...
+				<option value="5">J5</option>
+				<option value="10">J10</option>
+				<option value="15">J15</option>
+				<option value="20">J20</option>
+				<option value="25">J25</option>
+				<option value="30">J30</option>
+				<option value="50">J50</option>
+				<option value="100">J100</option>
+				<option value="0">JALL</option>
+			</field>
```
Comment (brianteeman) [inline components/com_finder/tmpl/search/default.xml]:
```suggestion
			<field
				name="list_limit"
				type="list"
				label="JGLOBAL_LIST_LIMIT"
				default="20"
				validate="options"
				>
				<option value="5">J5</option>
				<option value="10">J10</option>
				<option value="15">J15</option>
				<option value="20">J20</option>
				<option value="25">J25</option>
				<option value="30">J30</option>
				<option value="50">J50</option>
				<option value="100">J100</option>
				<option value="0">JALL</option>
			</field>
```

Code style and validation

**Comment by chmst** [comment]
This could be done for every list layout. 
But it is a new feature.
Documentation has to be changed if this is implemented.

**Comment by jiweigert** [comment]
Documentation change indeed,
but for the default value, I would recommend to get that value from prior source in the global configuration.
If someone have changed that value, it would be surprising when the number of results per page would change.

Maybe you could change this too Finder setting -> Global Setting -> Hardcoded value?
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31099">issues.joomla.org/tracker/joomla-cms/31099</a>.</sub>

**Comment by HLeithner** [comment]
@sakiss can you please merge the current 4.0-dev branch into your PR

**Comment by richard67** [comment]
> @sakiss can you please merge the current 4.0-dev branch into your PR

To be more precise: The 4.0-dev branch of the CMS repository, not of your fork. It seems your 4.0-dev branch was very outdated when you have created this PR and the other one, #31100 based on it, because both PRs fail in our automated test due to outdated testing configuration from before February.

**Comment by sakiss** [comment]
@HLeithner @richard67 Thank you guys!
I accidentally pulled from my origin repo instead of the upstream.

Though after pulling from the upstream, i am facing an exception `"There is no "field.passwordview" asset of a "script" type in the registry."` Run both` composer install` and `npm install` without any result.
Any help is welcome!

**Comment by richard67** [comment]
@sakiss Maybe not pull but merge changes from upstream/4.0-dev into origin/finder_pagination, assuming the remote for your GitHub repository (CMS fork) is "origin" and the remote for the CMS repository is "upstream".

You can check that by entering "git remote -vv" in a git client's command shell.

If the result looks as follows, it is like I assumed above:

```
origin  https://github.com/sakiss/joomla-cms.git (fetch)
origin  https://github.com/sakiss/joomla-cms.git (push)
upstream        https://github.com/joomla/joomla-cms.git (fetch)
upstream        https://github.com/joomla/joomla-cms.git (push)
```

A good way to always keep your base branches (staging, 4.0-dev, 4.1-dev) for making PRs up to date with the same branches of the CMS is e.g. for the 4.0-dev branch following sequence of commands in a git client's command shell:

```
git checkout 4.0-dev
git remote update
git clean -d -x -f
git reset --hard upstream/4.0-dev
push origin +4.0-dev
```

After this, the 4.0-dev branch of your GitHub repository will be identical with the 4.0-dev of the cms, and all local untracked files and folder, e.g. remainders of a previous Joomla installation using that git clone will be removed locally.

If then making a PR for 4.0, just make the branch for that PR based on your (previously updated as described above) 4.0-dev branch, and all will be fine.

For now I would suggest to try with merging as I wrote at the top.

**Comment by sakiss** [comment]
@richard67 Thank you!

My problem is that my Joomla is broken after pulling/merging from the upstream. `npm install` command seems like it did not complete and exited with me an error.
exception

**Comment by richard67** [comment]
As I've said first, you shall not pull but you shall merge into this branch.

**Comment by richard67** [comment]
Current 4.0-dev branch of the CMS works fine here, no such issue.

**Comment by VladimirAus** [comment]
I have tested this item :white_check_mark: successfully on 97bdb0cb25dbda51a3b7e4be88d8d621dcb460a3<br /><br />Worked. 
Dropdown for number of items appeared on the advanced menu tab after applying the patch.
Search results reflected the change.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31099">issues.joomla.org/tracker/joomla-cms/31099</a>.</sub>

**Comment by ThomasFinnern** [comment]
I have tested this item :white_check_mark: successfully on 97bdb0cb25dbda51a3b7e4be88d8d621dcb460a3<br /><br />Before 20 items, after 30 items found as expected<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31099">issues.joomla.org/tracker/joomla-cms/31099</a>.</sub>

**Comment by ThomasFinnern** [comment]
before 20 items after 30 items as expected<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31099">issues.joomla.org/tracker/joomla-cms/31099</a>.</sub>

**Comment by richard67** [comment]
@sakiss If you want, remove the draft status by using the "Ready for review" button. The PR has 2 good tests now and so could get RTC (ready to commit), but only if it is not draft anymore.

@HLeithner As you know about the issue with failing tests due to old branch: Would that keep us from merging it? I think no, but as usual I'm not 100% sure.

**Comment by HLeithner** [comment]
@richard67 I merged current 4.0-dev branched into the pr (only 1833 commits behind^^) drone should work now.

**Comment by richard67** [comment]
@HLeithner Appveyor shows it is not able to build non-mergeable pull request.

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**